Lib/python/pyrun.swg

```diff
@@ -1395,19 +1433,14 @@ SWIG_Python_SetModule(swig_module_info *swig_module) {
 #endif
   PyObject *pointer = PyCapsule_New((void *) swig_module, SWIGPY_CAPSULE_NAME, SWIG_Python_DestroyModule);
   if (pointer && module) {
-    PyModule_AddObject(module, "type_pointer_capsule" SWIG_TYPE_TABLE_NAME, pointer);
+    if (PyModule_AddObject(module, "type_pointer_capsule" SWIG_TYPE_TABLE_NAME, pointer) < 0) {
```

Comment:
```suggestion
    if (PyModule_AddObject(module, "type_pointer_capsule" SWIG_TYPE_TABLE_NAME, pointer) != 0) {
```

Reply:
The [documentation for `PyModule_AddObject`](https://docs.python.org/3/c-api/module.html?highlight=pymodule_addobject#c.PyModule_AddObject) has example usage which uses `< 0` rather than `!= 0` as the test:
```c
Py_INCREF(spam);
if (PyModule_AddObject(module, "spam", spam) < 0) {
    Py_DECREF(module);
    Py_DECREF(spam);
    return NULL;
}
```
Its protocol is described as:

> Return `-1` on error, `0` on success.

I compared this with the usage of `PyType_Ready()` in this file, whose [documentation states](https://docs.python.org/3/c-api/type.html?highlight=pytype_ready#c.PyType_Ready) that it will:

> Return `0` on success, or return `-1` and sets an exception on error.

Note the following use of `< 0` as the test with `PyType_Ready()` elsewhere in this file (on three occasions):
```c
    if (PyType_Ready(&swigpyobject_type) < 0)
      return NULL;
```
It therefore appeared to me to be most consistent to use `< 0` in this case, to match the existing use with `PyType_Ready()`.

Would you like me to:

1. Leave the code as-is
2. Take your change, and update calls to `PyType_Ready()` as well
3. Take your change, and but leave `PyType_Ready()` using `< 0`

?

Reply:
It's not that important, leave it, a consistent approach is good. I just don't think that `<0` correlates as well with the documented protocol and `!=0` generates nice and simply assembler.

Reply:
Thanks for the feedback - you would in general have me at "it generates simpler/better assembler"!

I've made the other changes to remove the unnecessary calls to `PyErr_NoMemory()` but have left this untouched. If you would like me to change to option 2, then I'd be happy to do this as well.